assembly/src/json.ts

```diff
@@ -559,6 +592,156 @@ export namespace JSON {
   return schema;
 }
 
+// @ts-ignore: Decorator
+@inline function parseMap<T extends Map>(data: string, initializeDefaultValues: boolean): T {
+
+  const map: nonnull<T> = changetype<nonnull<T>>(
+    __new(offsetof<nonnull<T>>(), idof<nonnull<T>>())
+  );
```

Comment:
Interestingly, `const map = instantiate<T>()` worked just fine for `yarn run test`, but failed to compile when calling `yarn run build:test`.  Not sure why, but this alternate approach works for both.

Also, FYI - there remiain three warnings when calling `yarn run build:test` that are unrelated to this PR.

Reply:
I checked on my end and it seems that those warnings are a result of using a newer AssemblyScript version. I'll update accordingly.
That's quite interesting. Not sure why thats failing to build